src/open_inwoner/react/lib/wc/wc.spec.tsx

```diff
@@ -0,0 +1,490 @@
+      })
+
+      it('should have props property', () => {
+        class TestWebComponent extends GenericReactWebComponent<any> {
+          constructor() {
+            super(TestComponent)
+          }
+        }
+
+        const elementName = 'test-props-element'
+        if (!customElements.get(elementName)) {
+          customElements.define(elementName, TestWebComponent)
+        }
+
+        const instance = document.createElement(
+          elementName
+        ) as any as TestWebComponent
+
+        expect(instance.props).toBeDefined()
+        expect(typeof instance.props).toBe('object')
+      })
+
+      it('should parse props from string attributes', () => {
+        class TestWebComponent extends GenericReactWebComponent<any> {
+          constructor() {
+            super(TestComponent)
+          }
+        }
+
+        const elementName = 'test-parse-element'
+        if (!customElements.get(elementName)) {
+          customElements.define(elementName, TestWebComponent)
+        }
+
+        // Create element with innerHTML to set attributes before construction
+        const div = document.createElement('div')
+        div.innerHTML = `<${elementName} message="Hello World"></${elementName}>`
+        const instance = div.querySelector(
+          elementName
+        ) as any as TestWebComponent
+
+        expect(instance.props.message).toBe('Hello World')
+      })
+
+      it('should convert kebab-case attributes to camelCase props', () => {
+        class TestWebComponent extends GenericReactWebComponent<any> {
+          constructor() {
+            super(TestComponent)
+          }
+        }
+
+        const elementName = 'test-kebab-element'
+        if (!customElements.get(elementName)) {
+          customElements.define(elementName, TestWebComponent)
+        }
+
+        const instance = document.createElement(
+          elementName
+        ) as any as TestWebComponent
+        instance.setAttribute('data-test-value', 'test')
+
+        // Create a new instance with attributes already set
+        const div = document.createElement('div')
+        div.innerHTML = `<${elementName} data-test-value="test"></${elementName}>`
+        const freshInstance = div.querySelector(
+          elementName
+        ) as any as TestWebComponent
+
+        expect(freshInstance.props.dataTestValue).toBe('test')
+      })
+
+      it('should parse JSON attribute values', () => {
+        class TestWebComponent extends GenericReactWebComponent<any> {
+          constructor() {
+            super(TestComponent)
+          }
+        }
+
+        const elementName = 'test-json-element'
+        if (!customElements.get(elementName)) {
+          customElements.define(elementName, TestWebComponent)
+        }
+
+        const jsonData = { key: 'value', nested: { prop: 42 } }
+        const div = document.createElement('div')
+        div.innerHTML = `<${elementName} data='${JSON.stringify(jsonData)}'></${elementName}>`
```

Comment:
The backend will HTML escape the JSON string. We should expand the test `jsonData` to include a bunch of strings that include HTML characters, and use something like https://lodash.info/doc/escape here to simulate what actually happens when Django passes JSON into the template:


```suggestion
        div.innerHTML = `<${elementName} data='${_.escape(JSON.stringify(jsonData))}'></${elementName}>`
```

Reply:
We don't use lodash yet in the project, but i could add it.